packages/metadata-editor/src/MetadataEditor.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FormGroup, MenuItem } from '@blueprintjs/core';
+import { ItemPredicate } from '@blueprintjs/select';
+import { FrontendServices } from '@elyra/application';
+import { ReactWidget } from '@jupyterlab/apputils';
+import { CodeEditor } from '@jupyterlab/codeeditor';
+import { Select, InputGroup, Button } from '@jupyterlab/ui-components';
+
+import { Message } from '@lumino/messaging';
+
+import * as React from 'react';
+
+const ELYRA_METADATA_EDITOR_CLASS = 'elyra-metadataEditor';
+
+const defaultLanguages = ['python', 'R', 'C#', 'scala'];
+
+/**
+ * Metadata editor widget
+ */
+export class MetadataEditor extends ReactWidget {
+  metadata: any;
+  newFile: boolean;
+  updateSignal: () => void;
+  editorFactory: CodeEditor.Factory;
+  editor: CodeEditor.IEditor;
+  endpoint: string;
+
+  constructor(
+    metadata: any,
+    newFile: boolean,
+    updateSignal: () => void,
+    editorFactory: CodeEditor.Factory | null,
+    endpoint: string
+  ) {
+    super();
+    this.metadata = metadata;
+    this.editorFactory = editorFactory;
+    this.endpoint = endpoint;
+    this.newFile = newFile;
+    this.updateSignal = updateSignal;
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+    this.handleLanguageChange = this.handleLanguageChange.bind(this);
+  }
+
+  onCloseRequest(msg: Message) {
```

Comment:
```suggestion
  onCloseRequest(msg: Message): void {
```
to fix the check warnings